kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -14374,26 +14374,45 @@ public void hardDelete(String siteId) {
 		}
 
 		// Cleanup the collections
-		String folderId = "";
-		try {
-			ContentCollection siteCollection = getCollection(collectionId);
-			List<ContentCollectionEdit> folders = m_storage.getCollections(siteCollection);
-			for (ContentCollectionEdit folder : folders) {
-				folderId = folder.getId();
-				log.debug("Removing collection: {}", folderId);
-				removeCollection(folder.getId());
-			}
-
-			// Now delete the site-root collection
-			log.debug("Removing collection: {}", siteCollection.getId());
-			removeCollection(siteCollection.getId());
-		} catch (IdUnusedException ide) {
-			log.warn("No resources in collection {}.", folderId);
-		} catch (Exception e) {
-			log.warn("Failed to remove collection {}.", folderId, e);
-		}
+		removeCollectionRecursive(collectionId);
     }
 
+	private void removeCollectionRecursive(String collectionId){
```

Comment:
Your indentation is off here, I think. Some of this content service code does have some wacky indentation going on. Also, you've collapsed a lot of white space. So, instead of if{, we try to use a space "if {". Same with all your try catch blocks. I'll leave a few commit suggestions with what I'm suggesting.